dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -277,53 +186,14 @@ export default defineComponent({
     },
 
     /**
-     * Checks for a single file name if it already occurs among the referenceable report names
-     * @param fullFileName is the full file name with its prefix that should be checked
-     * @returns a boolean stating if the file name is among the referenceable report names
-     */
-    isFileNameAlreadyExistingAmongReferenceableReportNames(fullFileName: string): boolean {
-      const fileNameWithoutSuffix = this.removeFileTypeExtension(fullFileName);
-      return this.allReferenceableReportNames.some((reportName) => reportName === fileNameWithoutSuffix);
-    },
-
-    /**
-     *  calculates the hash from a file
-     * @param [file] the file to calculate the hash for
-     * @returns a promise of the hash as string
+     * Checks if there is a report name twice in the list of referencable report names
+     * @returns a boolean stating if any file name is duplicated among the reference report names
      */
-    async calculateSha256HashFromFile(file: File): Promise<string> {
-      const buffer = await file.arrayBuffer();
-      const hashBuffer = await crypto.subtle.digest("SHA-256", buffer);
-      return this.toHex(hashBuffer);
-    },
-    /**
-     *  helper to encode a hash of type buffer in hex
-     * @param [buffer] the buffer to encode in hex
-     * @returns  the array as string, hex encoded
-     */
-    toHex(buffer: ArrayBuffer): string {
-      const array = Array.from(new Uint8Array(buffer)); // convert buffer to byte array
-      return array.map((b) => b.toString(16).padStart(2, "0")).join(""); // convert bytes to hex string
-    },
-
-    /**
-     * Removes the file extension after the last dot of the filename.
-     * E.g. someFileName.with.dots.pdf will be converted to someFileName.with.dots
-     * @param fileName the file name
-     * @returns the file name without the file extension after the last dot
-     */
-    removeFileTypeExtension(fileName: string): string {
-      return fileName.split(".").slice(0, -1).join(".");
+    duplicatesAmongReferenceableReportNames(): boolean {
+      return this.allReferenceableReportNames.length !== new Set(this.allReferenceableReportNames).size;
     },
   },
 });
-interface StoredReport extends CompanyReport {
-  reportName: string;
-}
-interface ReportToUpload extends CompanyReport {
-  fileForReport: File;
-  fileNameWithoutSuffix: string;
-}
 </script>
 
```

Comment:
Can with have a call in which you can shortly explain the changes in this component to me.

Reply:
Had the call